### ISSUE TITLE
[resolver] show possible candidates when resolving a version contrained requirement failed

### ISSUE THREAD

**Issue by laeubi** [bndtools/bnd#5516]
Assume I get the following error from the resolver:

```
[ERROR] Resolution failed. Capabilities satisfying the following requirements could not be found:
    [<<INITIAL>>]
      ⇒ osgi.identity: (&(osgi.identity=my.test.fragment)(version>=1.0.0))
          ⇒ [my.test.fragment version=1.0.0 type=osgi.fragment]
              ⇒ osgi.wiring.package: (osgi.wiring.package=org.osgi.test.junit5.service)
                  ⇒ [org.osgi.test.junit5 version=1.2.1]
                      ⇒ osgi.wiring.package: (&(osgi.wiring.package=org.osgi.framework)(version>=1.8.0)(!(version>=2.0.0)))
    [org.apache.felix.scr version=2.2.4]
      ⇒ osgi.wiring.package: (&(osgi.wiring.package=org.osgi.dto)(version>=1.1.0)(!(version>=2.0.0)))
```

the currently it is hard to guess what might be missing without carefully inspection of the used repositories, e.g.

- The package *might* exists, but the version do not match
- The package is not available at all.

It would be good if the resolver when it encounters some "well-known errors" (e.g. a missing package or identity ) it does the following:

1.  Query for the package/identity **without** a version, and print a lit of versions that could be found instead
2. If not package/identity is found, print that no available bundle export the package or no identity with that name is found

Even though this only works for same special cases, these are the most common ones and thus has a very great value to the user as errors in the setup are much easier to spot.

**Comment by laeubi** [comment]
I have now debugged this and the problem is that actually there *is* a provider in my repository but any Resource that has the "org.osgi.framework" package is excluded (in my case `org.eclipse.osgi`), as the `org.osgi.dto` is also exported by equinox this package is again "missing". The code is this one here:
https://github.com/bndtools/bnd/blob/2ec77552aec10777f187b45dfd028b7b5f399887/biz.aQute.resolve/src/biz/aQute/resolve/AbstractResolveContext.java#L319-L329

so it seems something in my setup for the resolver is not working as expected, but I think the idea is still good for other cases :-)

**Comment by kriegfrj** [comment]
To help debug your case, it might be nice if the resolver flagged where there were candidates available in resources that were excluded from consideration.

**Comment by laeubi** [comment]
> To help debug your case, it might be nice if the resolver flagged where there were candidates available in resources that were excluded from consideration.

Yes that would also be useful... in the meanwhile I found out that I used `-framework` instead of `-runfw` ... there is a error printout in the code but it does not showed up and did not failed anything (beside the strange resolver error).

**Comment by kriegfrj** [comment]
> ... in the meanwhile I found out that I used `-runframework` instead of `-runfw` ... there is a error printout in the code but it does not showed up and did not failed anything (beside the strange resolver error).

I've done that before. Like you, I found it frustratingly subtle and difficult to debug.

**Comment by pkriens** [comment]
I created a warning for the -runframework but the extended diagnostics will require someone with more time. Moved it to abeyance for this.
